robyn/__init__.py

```diff
@@ -427,33 +438,70 @@ class SubRouter(Robyn):
     def __init__(self, file_object: str, prefix: str = "", config: Config = Config()) -> None:
         super().__init__(file_object, config)
         self.prefix = prefix
+        self.sub_router = []
+        self.temp_routes = []  # Temporary storage for routes
+
+    def add_temp_route(self, route_type: HttpMethod, endpoint: str, handler: Callable, const: bool, auth_required: bool):
+        self.temp_routes.append((route_type, endpoint, handler, const, auth_required))
+
+    def include_router(self, router, added_routes=None):
+        """
+        The method to include the routes from another router
 
-    def __add_prefix(self, endpoint: str):
-        return f"{self.prefix}{endpoint}"
+        :param router Robyn: the router object to include the routes from
+        """
+        router.prefix = f"{self.prefix}{router.prefix}"
+        self.sub_router.append(router)
+        for sub_router in router.sub_router:
+            self.include_router(sub_router)
+
+    def get(self, endpoint: str, const: bool = False, auth_required: bool = False):
+        def inner(handler):
+            self.add_temp_route(HttpMethod.GET, endpoint, handler, const, auth_required)
```

Comment:
Also, why did you have to refactor the original implementation here? 

^^ I am trying to understand you thoughts here.

Reply:
The original implementation of the get, post, etc., decorators calls super(), which registers the route when the decorator is used. However, when nested subrouters are used, the decorator has already been called, so the route retains the old prefix instead of adopting the nested prefix.

Reply:
>The original implementation of the get, post, etc., decorators calls super(), which registers the route when the decorator is used. However, when nested subrouters are used, the decorator has already been called, so the route retains the old prefix instead of adopting the nested prefix.

Could you please add it in comment/docstring in the codebase?

Reply:
Of course. I'll add it.